Fix several tiny memory leaks

Quipyowert2 opened the pull request:
* **What does this PR do?**
Fixes a series of very tiny leaks.
* **Screenshots (if applicable)**

* **PR acceptance criteria** (reminder only, please delete once read)

  - Your commit message(s) are descriptive.  See:

    https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

  - [https://raw.githubusercontent.com/fvwmorg/fvwm3/master/doc/README]Documentation updated (where appropriate)

  - Style guide followed (try and match the surrounding code where possible)

  - All tests are passing:  although this is automatic, Codacy will often
    highlight additional considerations which will need to be addressed before
    the PR can be merged.

* **Issue number(s)**

If this PR addresses any issues, please ensure the appropriate commit
message(s) contains:

```
Fixes #XXX
```

at the end of your commit message, where `XXX` should be replaced with the
relevant issue number.

If there is more than one issue fixed then use:

```
Fixes #XXX, fixes #YYY, fixes #ZZZ
```

ThomasAdam commented on libs/FScreen.c:
```diff
@@ -409,9 +409,13 @@ scan_screens(Display *dpy)
 	for (i = 0; i < n; i++) {
 
 		char *name = XGetAtomName(dpy, rrm[i].name);
+		Bool must_free_name = True;
```
As an aside here, the "Bool" type is usually reserved for Xlib operations, and either "int" or "bool" (via `stdbool.h`) can be used instead.

ThomasAdam commented on libs/FScreen.c:
```diff
@@ -409,9 +409,13 @@ scan_screens(Display *dpy)
+		Bool must_free_name = True;
 
 		if (name == NULL)
+		{
 			name = "";
+			must_free_name = False;
+		}
 
```
Looking at this a bit more, if we can't detect a monitor name coming from the XAtom, then it's an error, and not something we should continue with.  In detecting monitors, we won't ever have stored the empty string to match against, so the rest of the function won't really do a great deal.  I'd rather see:

```
if (name == NULL) {
    fprintf(stderr, "%s: couldn't detect monitor with empty name\n", __func__);
    exit (101);
}
```

ThomasAdam commented on libs/FScreen.c:
```diff
@@ -447,9 +451,12 @@ scan_screens(Display *dpy)
 		m->virtual_scr.MyDisplayWidth = monitor_get_all_widths();
 		m->virtual_scr.MyDisplayHeight = monitor_get_all_heights();
 
+		if (must_free_name)
+			XFree(name);
```
No need to check to free here, given the above change.

ThomasAdam commented on libs/FScreen.c:
```diff
@@ -447,9 +451,12 @@ scan_screens(Display *dpy)
 	}
 
 	monitor_check_primary();
+	XRRFreeMonitors(rrm);
```
This is correct.

ThomasAdam reviewed (changes requested):
Hi @Quipyowert2 

Thanks, all look good.  Not overly wild about the whole "Bool" checking in functions.c, but there's already a precendence for it.

See comments in-line for the FScreen.c change.

Quipyowert2 commented:
I added the error message you suggested and removed the `must_free_name` variable since it wasn't needed. I squashed the commits into one commit but the commit message is kind of long.

ThomasAdam commented:
Thanks, @Quipyowert2 

I would use a commit message of:

```
Assorted mem leak fixes

In running fvwm3 though valgrind, this commit fixes a few memory leaks it has detected.
```

ThomasAdam commented on fvwm/virtual.c:
```diff
@@ -2383,7 +2388,7 @@ void CMD_GotoDeskAndPage(F_CMD_ARGS)
 	else
 		PeekToken(action, &action);
 
-	//free(action_cpy);
+	free(action_cpy_start);
```
Oops.  Thanks for fixing this!

Quipyowert2 replied:
`PeekToken` moves `action_cpy` forward a few bytes so I had to introduce another variable to be able to free the original pointer. Freeing `action_cpy` directly caused an Invalid free error from valgrind since it was pointing into the middle of the allocated memory instead of the beginning.

You're welcome.
